### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by crussell-figure**
## Description

* Updated proposal Monitor insert
* trying to make assetHolders() faster
* Fixed Smart Contract apis to actually return data
* Added requested NFT apis
* Updated existing NFT apis with additional data
* Started using Kotlin coroutines
* Started integrating with Kotlin protobuf supported functions
* Added Missed Blocks apis for devops
* Fixed tx_gas_fee_volume_* aggregations

closes: #253
closes: #254
closes: #255
closes: #257
closes: #258

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/explorer/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: proto/build.gradle.kts [hunk 1]
```diff
@@ -7,13 +7,15 @@ import com.google.protobuf.gradle.protoc
 
 sourceSets.main {
     proto.srcDirs("../third_party/proto/")
-    java.srcDirs("build/generated/source/proto/main/java")
 }
 
 dependencies {
-    api(Libraries.ProtobufJava)
-    api(Libraries.GrpcStub)
+//    protobuf(files("cosmWasm-v0.17.0.tar.gz"))
```
Comment (fkneeland-figure) [inline proto/build.gradle.kts:13]: Do we need this?

File: proto/build.gradle.kts [hunk 2]
```diff
@@ -51,3 +65,13 @@ tasks.register<io.provenance.DownloadProtosTask>("downloadProtos") {
     wasmdVersion = Versions.Wasmd
     ibcVersion = Versions.Ibc
 }
+
+//tasks.register("downloadTest"){
```
Comment (fkneeland-figure) [inline proto/build.gradle.kts:69]: Do we need this?

File: service/src/main/kotlin/io/provenance/explorer/grpc/v1/AccountGrpcClient.kt [hunk 3]
```diff
@@ -36,89 +43,81 @@ class AccountGrpcClient(channelUri: URI) {
                         it.usePlaintext()
                     }
                 }
-                .idleTimeout(60, TimeUnit.SECONDS)
-                .keepAliveTime(10, TimeUnit.SECONDS)
-                .keepAliveTimeout(10, TimeUnit.SECONDS)
+                .idleTimeout(5, TimeUnit.MINUTES)
```
Comment (fkneeland-figure) [inline service/src/main/kotlin/io/provenance/explorer/grpc/v1/AccountGrpcClient.kt:46]: Why are these timeouts being significantly increased?

**Review by fkneeland-figure — APPROVED** [review]

File: proto/build.gradle.kts (hunk 1 above)
Comment (crussell-figure) [reply proto/build.gradle.kts:13]: This is prep for when all protos will be included in the zip from provenance. I commented vs removing so I dont forget how to do it

File: proto/build.gradle.kts (hunk 2 above)
Comment (crussell-figure) [reply proto/build.gradle.kts:69]: See above

**Pull request merged**